pkg/testing/v1/service.go

```diff
@@ -172,6 +172,18 @@ func WithServiceAccountName(serviceAccountName string) ServiceOption {
 	}
 }
 
+// WithPodSpecDryRunEnabled enables the podspec dry-run feature.
+func WithPodSpecDryRunEnabled() ServiceOption {
+	return func(service *v1.Service) {
+		a := service.GetAnnotations()
+		if a == nil {
+			a = map[string]string{}
+			service.SetAnnotations(a)
+		}
+		a["features.knative.dev/podspec-dryrun"] = "enabled"
```

Comment:
We should make this a constant in `pkg/webhook` to be shared with the webhook code